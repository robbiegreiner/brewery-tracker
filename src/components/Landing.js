import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FeatureBrewery from './FeatureBrewery';
import FeatureBeer from './FeatureBeer';
import chevron from '../assets/chevron-sign-down.svg';


class Landing extends Component {
  constructor() {
    super();
    this.state = {
      searchValue: ''
    };
  }

  componentDidMount() {
    this.props.getFeatures();
    if (this.props.user) {
      this.props.getFavorites(this.props.user.id);
    }
  }

  handleChange(event) {
    this.setState({
      searchValue: event.target.value
    });
  }

  searchClick(type) {
    const { searchAll } = this.props;
    searchAll(this.state.searchValue, type);
  }

  searchCityState(searchType) {
    const { searchCity } = this.props;
    const cityAndState = this.state.searchValue.split(',');
    const city = cityAndState[0];
    let state = '';

    if (cityAndState[1]) {
      state = cityAndState[1];
    }
    searchCity(city, state, searchType);
  }

  render() {
    return (
      <div className='landing'>
        <div className='search-area'>
          <h1>Search for Beers & Breweries</h1>
          <div className='search-container'>
            <input placeholder="Search" onChange= {(event ) => this.handleChange(event)}></input>
            <Link to='/searchresults'>
              <button
                onClick={ () => this.searchClick('beer') }>Beers
              </button>
              <button
                onClick={ () => this.searchClick('brewery') }>Breweries
              </button>
              <button
                onClick={ () => this.searchCityState('city') }>City
              </button>
            </Link>
          </div>
        </div>
        <div className='feature-section'>
          <FeatureBrewery
            featureBrewery={this.props.featureBrewery}
            getBrewery ={this.props.getBrewery}/>
          <FeatureBeer
            featureBeer={this.props.featureBeer}
            getBrewery ={this.props.getBrewery}/>
        </div>
      </div>
    );

  }
}

Landing.propTypes = {
  searchAll: PropTypes.func,
  featureBrewery: PropTypes.object,
  featureBeer: PropTypes.object,
  getFeatures: PropTypes.func,
  getBrewery: PropTypes.func,
  searchCity: PropTypes.func,
  user: PropTypes.object,
  getFavorites: PropTypes.func
};

export default Landing;
