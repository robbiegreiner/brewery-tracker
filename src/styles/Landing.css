.landing {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center; }

.search-area {
  height: 100vh;
  width: 100vw;
  background: linear-gradient(90deg, rgba(49, 61, 65, 0.6), rgba(49, 61, 65, 0.6)), url("../assets/landing-img.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center; }
