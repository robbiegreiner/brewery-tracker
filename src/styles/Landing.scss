.landing {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.search-area {
  height: 82vh;
  width: 100vw;
  background:     linear-gradient( 90deg,
      rgba(49, 61, 65, 0.6),
      rgba(49, 61, 65, 0.6)
    ), url("../assets/landing-img.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.arrow-down {
  height: 35px;
  width: 35px;
  position: absolute;
  top: 90vh;
  z-index: 0;
}
