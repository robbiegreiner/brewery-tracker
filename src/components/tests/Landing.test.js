import React from "react";
import Landing from '../Landing';
import { shallow } from "enzyme";

describe('Landing component', () => {
  let wrapper;


  it('should match snapshot', () => {
    wrapper = shallow(<Landing />);
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });


});
