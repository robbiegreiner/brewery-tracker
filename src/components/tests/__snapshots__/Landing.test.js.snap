// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    featureBeer={Object {}}
    featureBrewery={Object {}}
    getBrewery={[Function]}
    getFeatures={[Function]}
    searchAll={[Function]}
    searchCity={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-area"
>
          <h1>
                    Search for Beers & Breweries
          </h1>
          <div
                    className="search-container"
          >
                    <input
                              onChange={[Function]}
                              placeholder="Search"
                    />
                    <Link
                              replace={false}
                              to="/searchresults"
                    >
                              <button
                                        onClick={[Function]}
                              >
                                        Beers
                              </button>
                              <button
                                        onClick={[Function]}
                              >
                                        Breweries
                              </button>
                              <button
                                        onClick={[Function]}
                              >
                                        City
                              </button>
                    </Link>
          </div>
</div>,
        <div
          className="feature-section"
>
          <FeatureBrewery
                    featureBrewery={Object {}}
                    getBrewery={[Function]}
          />
          <FeatureBeer
                    featureBeer={Object {}}
                    setCurrentBeer={undefined}
          />
</div>,
        <footer>
          <h3>
                    © Robbie Greiner 2017
          </h3>
</footer>,
      ],
      "className": "landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Search for Beers & Breweries
</h1>,
            <div
              className="search-container"
>
              <input
                            onChange={[Function]}
                            placeholder="Search"
              />
              <Link
                            replace={false}
                            to="/searchresults"
              >
                            <button
                                          onClick={[Function]}
                            >
                                          Beers
                            </button>
                            <button
                                          onClick={[Function]}
                            >
                                          Breweries
                            </button>
                            <button
                                          onClick={[Function]}
                            >
                                          City
                            </button>
              </Link>
</div>,
          ],
          "className": "search-area",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Search for Beers & Breweries",
            },
            "ref": null,
            "rendered": "Search for Beers & Breweries",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  onChange={[Function]}
                  placeholder="Search"
/>,
                <Link
                  replace={false}
                  to="/searchresults"
>
                  <button
                                    onClick={[Function]}
                  >
                                    Beers
                  </button>
                  <button
                                    onClick={[Function]}
                  >
                                    Breweries
                  </button>
                  <button
                                    onClick={[Function]}
                  >
                                    City
                  </button>
</Link>,
              ],
              "className": "search-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Search",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <button
                      onClick={[Function]}
>
                      Beers
</button>,
                    <button
                      onClick={[Function]}
>
                      Breweries
</button>,
                    <button
                      onClick={[Function]}
>
                      City
</button>,
                  ],
                  "replace": false,
                  "to": "/searchresults",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Beers",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Beers",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Breweries",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Breweries",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "City",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "City",
                    "type": "button",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FeatureBrewery
              featureBrewery={Object {}}
              getBrewery={[Function]}
/>,
            <FeatureBeer
              featureBeer={Object {}}
              setCurrentBeer={undefined}
/>,
          ],
          "className": "feature-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "featureBrewery": Object {},
              "getBrewery": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "featureBeer": Object {},
              "setCurrentBeer": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            © Robbie Greiner 2017
</h3>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "© Robbie Greiner 2017",
          },
          "ref": null,
          "rendered": "© Robbie Greiner 2017",
          "type": "h3",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-area"
>
            <h1>
                        Search for Beers & Breweries
            </h1>
            <div
                        className="search-container"
            >
                        <input
                                    onChange={[Function]}
                                    placeholder="Search"
                        />
                        <Link
                                    replace={false}
                                    to="/searchresults"
                        >
                                    <button
                                                onClick={[Function]}
                                    >
                                                Beers
                                    </button>
                                    <button
                                                onClick={[Function]}
                                    >
                                                Breweries
                                    </button>
                                    <button
                                                onClick={[Function]}
                                    >
                                                City
                                    </button>
                        </Link>
            </div>
</div>,
          <div
            className="feature-section"
>
            <FeatureBrewery
                        featureBrewery={Object {}}
                        getBrewery={[Function]}
            />
            <FeatureBeer
                        featureBeer={Object {}}
                        setCurrentBeer={undefined}
            />
</div>,
          <footer>
            <h3>
                        © Robbie Greiner 2017
            </h3>
</footer>,
        ],
        "className": "landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Search for Beers & Breweries
</h1>,
              <div
                className="search-container"
>
                <input
                                onChange={[Function]}
                                placeholder="Search"
                />
                <Link
                                replace={false}
                                to="/searchresults"
                >
                                <button
                                                onClick={[Function]}
                                >
                                                Beers
                                </button>
                                <button
                                                onClick={[Function]}
                                >
                                                Breweries
                                </button>
                                <button
                                                onClick={[Function]}
                                >
                                                City
                                </button>
                </Link>
</div>,
            ],
            "className": "search-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Search for Beers & Breweries",
              },
              "ref": null,
              "rendered": "Search for Beers & Breweries",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    onChange={[Function]}
                    placeholder="Search"
/>,
                  <Link
                    replace={false}
                    to="/searchresults"
>
                    <button
                                        onClick={[Function]}
                    >
                                        Beers
                    </button>
                    <button
                                        onClick={[Function]}
                    >
                                        Breweries
                    </button>
                    <button
                                        onClick={[Function]}
                    >
                                        City
                    </button>
</Link>,
                ],
                "className": "search-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "onChange": [Function],
                    "placeholder": "Search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <button
                        onClick={[Function]}
>
                        Beers
</button>,
                      <button
                        onClick={[Function]}
>
                        Breweries
</button>,
                      <button
                        onClick={[Function]}
>
                        City
</button>,
                    ],
                    "replace": false,
                    "to": "/searchresults",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Beers",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Beers",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Breweries",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Breweries",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "City",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "City",
                      "type": "button",
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FeatureBrewery
                featureBrewery={Object {}}
                getBrewery={[Function]}
/>,
              <FeatureBeer
                featureBeer={Object {}}
                setCurrentBeer={undefined}
/>,
            ],
            "className": "feature-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "featureBrewery": Object {},
                "getBrewery": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "featureBeer": Object {},
                "setCurrentBeer": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              © Robbie Greiner 2017
</h3>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "© Robbie Greiner 2017",
            },
            "ref": null,
            "rendered": "© Robbie Greiner 2017",
            "type": "h3",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
